Fixed Reading List on Mobile + Small Improvements

nickytonline opened the pull request:
<details>
<summary>Details</summary>



</details>

<!--
     For Work In Progress Pull Requests, please use the Draft PR feature,
     see https://github.blog/2019-02-14-introducing-draft-pull-requests/ for further details.

     For a timely review/response, please avoid force-pushing additional
     commits if your PR already received reviews or comments.

     Before submitting a Pull Request, please ensure you've done the following:
     - 📖 Read the Forem Contributing Guide: https://github.com/forem/forem/blob/master/CONTRIBUTING.md#create-a-pull-request.
     - 📖 Read the Forem Code of Conduct: https://github.com/forem/forem/blob/master/CODE_OF_CONDUCT.md.
     - 👷‍♀️ Create small PRs. In most cases this will be possible.
     - ✅ Provide tests for your changes.
     - 📝 Use descriptive commit messages.
     - 📗 Update any related documentation and include any relevant screenshots.

     NOTE: Pull Requests from forked repositories will need to be reviewed by
     a Forem Team member before any CI builds will run. Once your PR is approved
     with a `/ci` reply to the PR, it will be allowed to run subsequent builds without
     manual approval.
-->

## What type of PR is this? (check all applicable)

- [x] Refactor
- [ ] Feature
- [x] Bug Fix
- [ ] Optimization
- [ ] Documentation Update

## Description

Implements changes proposed in #11983. One notable change is that [we only filter by one tag at a time now](https://github.com/forem/forem/issues/11033#issuecomment-717087477).

Before

https://user-images.githubusercontent.com/833231/110536113-14d56300-80ef-11eb-9586-5b085c17b0c2.mp4

After

https://user-images.githubusercontent.com/833231/113223886-2a562c80-9258-11eb-9caf-b6497f2073ef.mp4

## Related Tickets & Documents

Closes #11983

## QA Instructions, Screenshots, Recordings

The reading list section should still be able to archive and unarchive reading list items.

### Archiving

1. Navigate to http://localhost:3000/
2. In the feed, click on the save button of an article in the feed to add it to the reading list.
2. Navigate to http://localhost:3000/readinglist
2. Click on the _Archive_ button of the reading list item.
4. The archived item will disappear from the reading list.
5. A snackbar item at the bottom left of the screen should say "Archiving..." and then disappear after a second or two.

### Unarchiving

1. Navigate to http://localhost:3000/
2. In the feed, click on the save button of an article in the feed to add it to the reading list.
2. Navigate to http://localhost:3000/readinglist
3. Click on the _Archive_ button of the reading list item.
4. The unarchived item will disappear from the archived reading list.
5. A snackbar item at the bottom left of the screen should say "Archiving..." and then disappear after a second or two.
6. Navigate to http://localhost:3000/readinglist/archive
7. Click on the _Unarchive_ button of the archived reading list item.
8. A snackbar item at the bottom left of the screen should say "Unarchiving..." and then disappear after a second or two.

### Filtering

#### Filtering the Reading List

1. Navigate to http://localhost:3000/
2. In the feed, click on the save button of an article in the feed to add it to the reading list. Do this with 4 or 5 articles so the reading list is a small size.
3. Navigate to http://localhost:3000/readinglist
4. In the search text box, enter in some text that is in the title of an article. It will filter so only article titles with that text are visible.
5. In the filter by tag drop down, select a tag. It will filter so only article titles with the current filter text and articles with the selected tag are visible.
6. Clear the search text box. It will filter so only articles with the selected tag are visible.

#### Filtering the Archive

To test the same functionality in the archived view:

1. From the reading list with the existing list from the previous test set up, click on the archive button of an item in the reading list for as many articles as you'd like.
2. Repeat steps 4-6 from the "Filtering the Reading List" section of QA.

### UI accessibility concerns?

I fixed the markup so that everything is contained within the `<main />` landmark now. There is a colour contrast issue with the `<select />` being reported, but this is not related to the work in the PR. This appears to be more of a global issue with `<select />`s. cc: @ludwiczakpawel 

@aitchiss, the fieldset on the left when on the medium breakpoint and up is now a group of radio buttons since we only allow for filtering on one tag now. The keyboard navigation appears to be working well as is Voice Over, but please take it for a spin to see if there are any gaps in the work there.

## Added tests?

- [x] Yes
- [ ] No, and this is why: _please replace this line with details on why tests
      have not been included_
- [ ] I need help with writing tests

## [Forem core team only] How will this change be communicated?

_Will this PR introduce a change that impacts Forem members or creators, the
development process, or any of our internal teams? If so, please note how you
will share this change with the people who need to know about it._

- [ ] I've updated the [Developer Docs](https://docs.forem.com) and/or
      [Admin Guide](https://forem.gitbook.io/forem-admin-guide/), or
      [Storybook](https://storybook.forem.com/) (for Crayons components)
- [ ] I've updated the README or added inline documentation
- [x] I will share this change in a [Changelog](https://forem.dev/t/changelog)
      or in a [forem.dev](http://forem.dev) post
- [ ] I will share this change internally with the appropriate teams
- [ ] I'm not sure how best to communicate this change and need help
- [ ] This change does not need to be communicated, and this is why not: _please
      replace this line with details on why this change doesn't need to be
      shared_

## [optional] Are there any post deployment tasks we need to perform?

## [optional] What gif best describes this PR or how it makes you feel?

![RHCP on the Simpsons where Flea says "Wow! That's much better. Everyone can enjoy that."](https://media.giphy.com/media/l2JeiFRnroKCfCEVi/giphy.gif)

nickytonline commented:
I've scrapped some component tests that existed for this and will write some end to end tests instead.

ludwiczakpawel commented on app/javascript/readingList/components/TagList.jsx:
```diff
@@ -0,0 +1,49 @@
+        <option
+          selected={tag === selectedTag}
+          className={`crayons-link crayons-link--block ${
+            tag === selectedTag ? 'crayons-link--current' : ''
```
Out of curiosity, why not this:

```
tag === selectedTag && 'crayons-link--current'
```

^ since there's no alternative value anyway..?

nickytonline replied:
If you do it this way, it will add a CSS class name called "false".

ludwiczakpawel replied:
😲 javascript is fun...

ludwiczakpawel commented on app/javascript/readingList/readingList.jsx:
```diff
@@ -13,32 +12,49 @@ import {
 import { ItemListItem } from './components/ItemListItem';
 import { ItemListItemArchiveButton } from './components/ItemListItemArchiveButton';
```
We should at some point look into components names: `ItemListItem` is probably not very descriptive...

ludwiczakpawel commented on app/javascript/readingList/components/TagList.jsx:
```diff
@@ -0,0 +1,49 @@
+    <select
+      class="crayons-select"
+      aria-label="Filter by tag"
+      onBlur={(event) => {
+        // We need blur for a11y, but we also don't want to make the same search query twice
+        // if the tag hasn't changed since the previous onChange.
+        const { value } = event.target;
+
+        if (value === selectedTag) {
+          return;
+        }
+
+        onSelectTag(event);
+      }}
```
Can you help me understand what does it do? I couldn't notice any difference when I removed this part of code.

ludwiczakpawel reviewed (changes requested):
I know it's a Draft PR but you added me as reviewers so here I am :D

1. Desktop UI regression. This one is IMO a blocker - afair we didn't want to replace sidebar nav with `select` field everywhere but only on smaller breakpoint(s). Bigger breakpoints should keep the left sidebar. This is similar to what we do on notifications and settings and that's likely the pattern we want to keep elsewhere.
    ![image](https://user-images.githubusercontent.com/108287/108679339-a8eadc00-74ec-11eb-9f0a-4803833e9537.png)

2. Going back to main view - blocker as well. When I choose a tag I can no longer go back to "main" view of Reading list - without any tag selected. That should probably be its own option in the select like "Everything" or something..
3. a11y issue - blocker too. When I use keyboard to choose a tag in this select, then hit `Shift + Tab` to go to previous field (search field), the filtering is being reset:
    
    https://user-images.githubusercontent.com/108287/108681811-e2711680-74ef-11eb-91d5-a8278d10c9a8.mov

    It's probably because search field, even if empty, is running a call and returns results matching the query (query is empty so it return everything). In other words, we should probably limit search field to only search through current view (selectedTag).

---

Not a blocker but something to think about in future is being able to reflect the current filter with URL. In other words, when you change the filtering we could update the browser's URL too to `dev.to/readinglist/javascript`. I'm aware it's something we don't have right now so there's no expectation to build it :D But you know, just something to keep in back of our heads.

nickytonline commented:
Ahh, I must have misread the comment about select. I thought we wanted it for desktop as well.

I'll look at the other issues as well you mentioned.

In regards to keeping the filter in the URL, I completely agree, but that should be a separate PR a that changes how the entire state is managed for this component.

ludwiczakpawel commented:
@nickytonline just to make sure I'm not making things up right now :D: 

> **Expected behavior**
> ...
> 2. Collapse nav into the select on mobile breakpoint.

^ that's also being mentioned in other comments in the same [issue](https://github.com/forem/forem/issues/11033).

nickytonline commented:
@ludwiczakpawel, @lisasy I've reworked this. The sidebar remains until we hit the smallest breakpoint. I took some liberties from some minor spacing changes, so let me know what you think.

@aitchiss, I also reworked the side bar to be a group of checkboxes even though visually it looks the same. It helps with the keyboard nav and voice over.

Here is the loom video, https://www.loom.com/share/c78e3427cfdf4b5ba01506ddd01ace32

Tests are still broken, but I will sort that out.

nickytonline commented:
I'm putting this on hold after I'm back as I'm looking into a higher priority issue in tegards to service workers, but if you can take this for a test drive @lisasy, @ludwiczakpawel, and @aitchiss, it'd be much appreciated. The only blocker for this PR at the moment is tests.

nickytonline commented:
As mentioned, I'll resume this when I'm back from my time off, but I put up some e2e tests @aitchiss. I discovered that Cypress is not evaluating `window.matchMedia` properly and I think it's related to https://github.com/cypress-io/cypress/issues/311. The browser API is working, but it always returns false when trying to match breakpoints.

ludwiczakpawel commented:
@nickytonline just turn it to "Ready for review" when ready :)

lisasy commented:
I can't seem to review this PR on my local dev, so my feedback is based on the video in the description. On the smallest breakpoint, there ought to be appropriate margin around the heading, search input, dropdown, and button so that these components aren't touching the edges of the device.

nickytonline commented:
> I can't seem to review this PR on my local dev, so my feedback is based on the video in the description. On the smallest breakpoint, there ought to be appropriate margin around the heading, search input, dropdown, and button so that these components aren't touching the edges of the device.

That seems to be the way the crayon layout class works, but I can definitely add margin. Also, those videos are out of date as that was my initial pass. I'll be posting updated videos shortly and will pull this out of draft mode.

nickytonline commented on app/javascript/readingList/readingList.jsx:
```diff
@@ -73,11 +91,11 @@ export class ReadingList extends Component {
     this.search(query, {
       page: 0,
-      tags: selectedTags,
+      tags: [],
```
Although we're filtering by one tag now, the search query still requires an array which is why this gets reset to an empty array.

nickytonline commented on app/javascript/readingList/readingList.jsx:
```diff
@@ -210,30 +217,65 @@ export class ReadingList extends Component {
+        </header>
+        <MediaQuery
+          query={`(width >= ${BREAKPOINTS.Medium}px)`}
+          render={(matches) => {
```
@aitchiss as [per your question in the PR for creating the `<MediaQuery />`](https://github.com/forem/forem/pull/12809#discussion_r582669298) component, this is why I opted for a render prop.

nickytonline commented on cypress/fixtures/search/readingList.json:
```diff
@@ -0,0 +1,152 @@
+{
```
Reading list mocked data for e2e tests.

nickytonline commented on jest.config.js:
```diff
@@ -25,8 +25,8 @@ module.exports = {
     global: {
-      statements: 43,
-      branches: 39,
+      statements: 42,
```
I reduced the coverage thresholds because the component tests for the reading list no longer exist. I've converted them to E2E tests. @djuber, this is something I haven't had time to look into/do yet, but it'd be good to aggregate the test coverage from jest with that of Cypress so that we can get a true picture of the code coverage for the front-end.

nickytonline commented:
There are some flashes of content that I'm going to look into which existed previously. It might merit a separate PR though.


https://user-images.githubusercontent.com/833231/110524551-8ad2cd80-80e1-11eb-9b13-8ed1829f19b1.mp4

nickytonline commented:
Just sorting out some E2E test failures...

aitchiss reviewed (commented):
In Chrome I'm not seeing the tag select on the reading list page, and the desktop layout has gone funky (using a private browsing window)...

![screenshot showing mobile layout with no tag filter](https://user-images.githubusercontent.com/20773163/110622303-35ef8f80-8193-11eb-8873-0974d4eb9496.png)

![screenshot showing desktop view with content rendered in small space to the side](https://user-images.githubusercontent.com/20773163/110622338-3d169d80-8193-11eb-8f7c-d44682844c88.png)

In Firefox it's fine though - maybe an issue with the media query in Chrome?

nickytonline commented:
> In Chrome I'm not seeing the tag select on the reading list page, and the desktop layout has gone funky (using a private browsing window)...
> 
> ![screenshot showing mobile layout with no tag filter](https://user-images.githubusercontent.com/20773163/110622303-35ef8f80-8193-11eb-8873-0974d4eb9496.png)
> 
> ![screenshot showing desktop view with content rendered in small space to the side](https://user-images.githubusercontent.com/20773163/110622338-3d169d80-8193-11eb-8f7c-d44682844c88.png)
> 
> In Firefox it's fine though - maybe an issue with the media query in Chrome?

Hmm, I'll take a peek as I've been working in Firefox lately. Seemed fine when running E2E tests in Chrome though. 🤔

nickytonline commented:
@aitchiss, I fixed the issue in Chrome. Bad media queries on my end. It also removed an issue I thought I had with media queries in Cypress, so I no longer need to mock `matchMedia` 🆒

ludwiczakpawel commented on app/javascript/readingList/components/TagList.jsx:
```diff
@@ -0,0 +1,88 @@
+
+function LargeScreenTagList({ availableTags, selectedTag, onSelectTag }) {
+  return (
+    <fieldset className="hidden grid grid-cols-1 gap-2">
+      <legend className="hidden">Filter by tag</legend>
+      {availableTags.map((tag) => (
+        <label
+          className={`crayons-link crayons-link--block${
+            selectedTag === tag ? ' crayons-link--current' : ''
+          }`}
+          aria-label={`${tag} tag`}
+        >
+          <input
+            type="radio"
+            name="filterTag"
+            onClick={onSelectTag}
+            key={tag}
+            className="opacity-0"
+            checked={selectedTag === tag}
+            value={tag}
+          />
+          #{tag}
+        </label>
+      ))}
+    </fieldset>
```
Using radio input here is a nice hack but it makes the nav accessibility a bit weird: when using my keyboard [tab] it focus the first element only and then jumps straight to articles list.. when going back [shift+tab] it highlights the last element only... Since this is radio, I'd have to use arrows to navigate through these radio options but that seems a bit confusing.

https://user-images.githubusercontent.com/108287/110937032-15117080-8332-11eb-9863-2208ecf2f68f.mov

(I added red outline to highlight focused elements in my vide but for some reason it didn't apply to tags :P but i hope you can spot what I'm talking about)

My suggestion would be replacing radio elements with classic links (or at least buttons?) but I can imagine it will require a bit more refactoring. But having form elements here instead of regular links doesn't seem to be the right choice.

ludwiczakpawel replied:
Oh, I've just noticed your older comment about refactoring that in a separate PR. Sure. 

> In regards to keeping the filter in the URL, I completely agree, but that should be a separate PR a that changes how the entire state is managed for this component.

^ I agree but in this case I wouldn't refactor the nav to be a form radio inputs (they currently are links in prod), since we will circle back to this anyway and turn them to be links again - instead probably only make adjustments to the `onSelectTag` function so we can avoid that UX regression with keyboard navigation introduced by turning links into form elements.

aitchiss replied:
I agree with @ludwiczakpawel I think for a sighted user it's not clear how to navigate the tags when they don't look like radios. 

Thinking about it, at the moment it's almost behaving more like a set of Tabs, with a single active page/tab at a time, but that also doesn't feel intuitive with the UI. It's been suggested already, but making these links and having the filter in the URL makes a lot of sense, and when we do that work this list can become a secondary `nav`.

tl;dr I second @ludwiczakpawel's suggestion to keep them as they are until we make that change.

nickytonline replied:
I can put it back and disable the eslint rule that complains we're using a link like a button. Like already mentioned, the links in prod have click event listeners on them already.

ludwiczakpawel reviewed (changes requested):
Couple things I've noticed:

1. There's that relayout jump on the page load - I think we should _always_ display sidebar even if there's no tags (which in most cases will happen only when there're no saved articles). 
    https://user-images.githubusercontent.com/108287/110935256-7c79f100-832f-11eb-86a4-8ea900d48f72.mov
2. On the left sidebar I should be able to "reset" my view - so I can see all saved articles again.. Probably having a link like "Everything" would be enough. (that's also missing in the mobile select)
    ![image](https://user-images.githubusercontent.com/108287/110935449-b3e89d80-832f-11eb-8ba8-0206ba6e898c.png)
3. If there're no saved articles, we can probably hide select on mobile. To avoid "relayout jump", when there are articles, we can show this select on the same line as "View archive" button maybe?
4. This wide "View archive" button seems a bit off. Maybe instead of using `grid` in header (which causes the issue in the first place) you could use flexbox? 
    ![image](https://user-images.githubusercontent.com/108287/110936085-a08a0200-8330-11eb-9747-d540a6d9a542.png)

ludwiczakpawel commented:
Oh, and that is still an issue:

> 3. When I use keyboard to choose a tag in this select, then hit Shift + Tab to go to previous field (search field), the filtering is being reset:

https://user-images.githubusercontent.com/108287/110938304-f44a1a80-8333-11eb-848a-bc06fe60c0df.mov

aitchiss commented on app/javascript/readingList/components/TagList.jsx:
```diff
@@ -0,0 +1,88 @@
+ * @param {object} props
+ * @param {Array<string>} props.availableTags A list of available tags.
+ * @param {string} [props.selectedTag=''] The currently selected tag.
+ * @param {function} A handler for when the selected tag changes.
```
```suggestion
 * @param {function} props.onSelectTag A handler for when the selected tag changes.
```
Also we should add the `isMobile` param

aitchiss commented on app/javascript/readingList/components/TagList.jsx:
```diff
@@ -0,0 +1,88 @@
+      class="crayons-select"
+      aria-label="Filter by tag"
+      onBlur={(event) => {
+        // We need blur for a11y, but we also don't want to make the same search query twice
+        // if the tag hasn't changed since the previous onChange.
```
I don't think we do need it anymore - the linter rule for this is deprecated as the issues were reported on lower than IE 10 https://github.com/jsx-eslint/eslint-plugin-jsx-a11y/blob/master/docs/rules/no-onchange.md

nickytonline replied:
I turned off this rule in #12986.

aitchiss commented on app/javascript/readingList/readingList.jsx:
```diff
@@ -186,17 +193,17 @@ export class ReadingList extends Component {
     return (
-      <section>
-        <header className="crayons-layout flex justify-between items-center pb-0">
+      <main>
```
this is a small thing but it makes me very happy 😄

aitchiss commented on docs/tests/e2e-tests.md:
```diff
@@ -108,6 +108,42 @@ Pro as there were issues integrating the cypress-rails gem with the
+## Custom Cypress commands
+
+[Cypress custom commands](https://docs.cypress.io/api/cypress-api/custom-commands.html)
+allow you do extend the functionality of the E2E testing framework. In the case
```
```suggestion
allow you to extend the functionality of the E2E testing framework. In the case
```

nickytonline commented:
> Couple things I've noticed:
> 
>     1. There's that relayout jump on the page load - I think we should _always_ display sidebar even if there's no tags (which in most cases will happen only when there're no saved articles).

Fixed. The sidebar is always displayed now in the wide view. It actually makes sense now that there is also a link to show all articles.

![image](https://user-images.githubusercontent.com/833231/111723014-4d84e300-8839-11eb-852d-654e2026729d.png)

![image](https://user-images.githubusercontent.com/833231/111723050-5c6b9580-8839-11eb-85a6-32b1ca6cdd85.png)


>     2. On the left sidebar I should be able to "reset" my view - so I can see all saved articles again.. Probably having a link like "Everything" would be enough. (that's also missing in the mobile select)

There is now the `All tags` option.

![image](https://user-images.githubusercontent.com/833231/111723119-6f7e6580-8839-11eb-998b-e9557321890b.png)

>     3. If there're no saved articles, we can probably hide select on mobile. To avoid "relayout jump", when there are articles, we can show this select on the same line as "View archive" button maybe?

Given that we have the All tags option now, the select would only have that option, or do you think it makes more sense to hide it still and also in the wide view to not render the `All tags` link?

>     4. This wide "View archive" button seems a bit off. Maybe instead of using `grid` in header (which causes the issue in the first place) you could use flexbox?

Taking a peek at that.

nickytonline commented:
Alright, finally got back to this PR. This should be in a good state to check out now @ludwiczakpawel and @aitchiss.

nickytonline commented on app/javascript/readingList/__tests__/readingList.test.jsx:
```diff
@@ -62,16 +62,4 @@ describe('<ReadingList />', () => {
     expect(results).toHaveNoViolations();
   });
-
-  it('renders all the elements', () => {
```
Covered by the E2E tests now.

ludwiczakpawel commented on app/javascript/readingList/readingList.jsx:
```diff
@@ -187,17 +193,17 @@ export class ReadingList extends Component {
           </h1>
-
-          <div class="flex items-center">
+          <fieldset className="grid gap-2 m:flex m:justify-end l:mb-0 mb-2 px-2 m:px-0">
```
If you add `m:items-center` it will better display button and input (right now, it's kinda enlarged).

nickytonline replied:
Done.

ludwiczakpawel reviewed (changes requested):
Much better mate, thanks! It's almost good to go:)

There're still two issues:

1. Filter by tag, then "tab back" to focus on search field (but don't type anything) then hit `Shift + Tab` again (to focus on the "View archive" button - it will reset the tags filtering.
2. Every time I click through the tags in left sidebar, I can see "empty state" for split of a second. Can we not have it while fetching the results? I don't even think we need to show a spinner since it happens almost instantly. 

Also, one tiny suggestion in code.

nickytonline commented:
> Much better mate, thanks! It's almost good to go:)
> 
> 
> 
> There're still two issues:
> 
> 
> 
> 1. Filter by tag, then "tab back" to focus on search field (but don't type anything) then hit `Shift + Tab` again (to focus on the "View archive" button - it will reset the tags filtering.
> 
> 2. Every time I click through the tags in left sidebar, I can see "empty state" for split of a second. Can we not have it while fetching the results? I don't even think we need to show a spinner since it happens almost instantly. 
> 
> 
> 
> Also, one tiny suggestion in code.

For 2., it was already like that. I was trying to keep the scope of the PR to just making things responsive, butI can probably squeak that in. 😎

ludwiczakpawel commented:
> For 2., it was already like that. I was trying to keep the scope of the PR to just making things responsive, butI can probably squeak that in. 😎

You're right! Yea, if you find it being a quick fix would be nice to do it. If not, that's absolutely fine then :)

nickytonline commented:
> Filter by tag, then "tab back" to focus on search field (but don't type anything) then hit Shift + Tab again (to focus on the "View archive" button - it will reset the tags filtering.

Fixed. This was a regression caused by moving to filtering by one tag only @ludwiczakpawel.

Going to look at fixing the empty state issue and some tests and this will be good to go for hopefully a final review. 😅

aitchiss commented on app/javascript/readingList/__tests__/readingList.test.jsx:
```diff
@@ -54,7 +65,7 @@ describe('<ReadingList />', () => {
   });
 
-  it('should have no a11y violations', async () => {
+  it('should have no a11y violations on small screens', async () => {
```
nitpick - this test and the "should have no a11y violations on large screens" looks to be the exact same?

nickytonline replied:
Oops, will fix that.

nickytonline replied:
done

aitchiss commented on app/javascript/readingList/components/TagList.jsx:
```diff
@@ -0,0 +1,81 @@
+function LargeScreenTagList({ availableTags, selectedTag, onSelectTag }) {
+  return (
+    <nav aria-label="Filter by tag" className="hidden grid grid-cols-1 gap-2">
+      <a
+        className={`crayons-link crayons-link--block${
+          !selectedTag ? ' crayons-link--current' : ''
+        }`}
+        data-no-instant
+        onClick={onSelectTag}
+        href="/t"
+      >
+        all tags
+      </a>
+      {availableTags.map((tag) => (
+        <a
+          className={`crayons-link crayons-link--block${
+            selectedTag === tag ? ' crayons-link--current' : ''
+          }`}
+          data-no-instant
+          data-tag={tag}
+          onClick={onSelectTag}
+          key={tag}
+          href={`t/${tag}`}
+        >
+          #{tag}
+        </a>
+      ))}
```
refactor (blocking) - can we make these links list items inside a `<ul>`?

nickytonline replied:
done

aitchiss commented on app/javascript/readingList/components/TagList.jsx:
```diff
@@ -0,0 +1,81 @@
+ * @param {object} props
+ * @param {Array<string>} props.availableTags A list of available tags.
+ * @param {string} [props.selectedTag=''] The currently selected tag.
+ * @param {function} props.onSelectTag A handler for when the selected tag changes.
```
nitpick (blocking) - we're missing the `isMobile` prop here

nickytonline replied:
done

aitchiss commented on app/javascript/readingList/readingList.jsx:
```diff
@@ -187,17 +194,17 @@ export class ReadingList extends Component {
     return (
-      <section>
-        <header className="crayons-layout flex justify-between items-center pb-0">
+      <main>
```
suggestion (not-blocking) - if we add the id `main-content` here, the skip link would work. Could also be picked up in a separate PR though as it's not really related to this issue

```suggestion
      <main id="main-content">
```

aitchiss commented on app/javascript/readingList/readingList.jsx:
```diff
@@ -206,35 +213,70 @@ export class ReadingList extends Component {
             </Button>
             <input
-              aria-label="Search..."
+              aria-label="Filter reading list by text"
```
so much better 😄

aitchiss reviewed (changes requested):
Checked this locally on desktop and android mobile and all looking good - I just have some v minor change requests/suggestions 🙂

ludwiczakpawel reviewed (approved):
NICE!

nickytonline commented:
I also tested this in iOS 14.4.2 and all looks good.

aitchiss reviewed (approved):
Nice! 🚀
